lib/chef/mixin/params_validate.rb

```diff
@@ -81,34 +81,58 @@ def lazy(&block)
         DelayedEvaluator.new(&block)
       end
 
-      def set_or_return(symbol, arg, validation)
+      NULL_ARG = Object.new
+
+      def nillable_set_or_return(symbol, arg, validation)
         iv_symbol = "@#{symbol.to_s}".to_sym
-        if arg == nil && self.instance_variable_defined?(iv_symbol) == true
-          ivar = self.instance_variable_get(iv_symbol)
-          if(ivar.is_a?(DelayedEvaluator))
-            validate({ symbol => ivar.call }, { symbol => validation })[symbol]
+        if arg == NULL_ARG
```

Comment:
for maximum ruby pedantry you might want to do `NULL_ARG.equal?(arg)` instead. `arg == other` behavior is defined by how `arg` implements `==` which could possibly (not super likely) be insane. Also this is a rare case where using object equality is exactly what you want (even though for Object, `==` and `eql?` are just aliases for `equal?`).